BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/location/provider/DeviceLocationPredictor.java

```diff
@@ -0,0 +1,137 @@
+package com.nexenio.bleindoorpositioning.location.provider;
+
+import com.nexenio.bleindoorpositioning.location.Location;
+import com.nexenio.bleindoorpositioning.location.distance.LocationDistanceCalculator;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Created by leon on 29.01.18.
+ * Stores window of last locations and tries to predict the next
+ * location based on average distance and angle during the window
+ */
+
+public class DeviceLocationPredictor {
+
+    private List<Location> pastDeviceLocations;
+    private Location predictedLocation;
+    public boolean hasPrediction;
+
+    public DeviceLocationPredictor() {
+        pastDeviceLocations = new ArrayList<>();
```

Comment:
Initialize when declaring

Reply:
Causes pastLocations to be overwritten with new ArrayList